src/components/Launch/components/LaunchImage/index.js

```diff
@@ -4,36 +4,46 @@ import "./styles.scss";
 export class LaunchImage extends React.Component {
     constructor(props) {
         super(props);
-        this.state = {//need to add derivedstatefromprops(nextProps, prevState)
-            showImages: this.props.imgBool || false
+        this.state = {
+            isShowingImages: this.props.isShowingImages || false
         }
     }
-    showImagesUpdate = () => {
+    isShowingImagesUpdate = () => {
         this.setState({
-            showImages: !this.state.showImages
+            isShowingImages: !this.state.isShowingImages
         });
     }
+
+    static getDerivedStateFromProps(nextProps, prevState){
+        if(nextProps.isShowingImages!==prevState.isShowingImages){
+            return {isShowingImages: nextProps.isShowingImages}
+        }
+        else{
+            return prevState;
+        }
+    }
     render() {
 
         return (
             <>
                 <img src={this.props.links.mission_patch}
                     alt="mission patch"
                     className="launchImg"></img>
-                {this.props.links.flickr_images.length > 0 &&
+                {(this.state.isShowingImages && this.props.isCurrentRoute===false) &&
                     <button className="showImagesButton"
-                        onClick={this.showImagesUpdate}>
+                        onClick={this.isShowingImagesUpdate}>
                         Show Images</button>}
                 
-                {this.state.showImages &&
+                {this.state.isShowingImages &&
+                    (this.props.links.flickr_images.length > 0 ?
                     <div className="flickr">
                     {this.props.links.flickr_images.map((img, index) => {
                         return <img src={img}
                             alt={"flickr " + index}
                             className="flickrImg"
                             key={"flickr " + index}></img>;
```

Comment:
React prefers self-closing tags when there are no children. So for this image, you can use `<img ... />`.